src/components/Confirm/Confirm.vue

```diff
@@ -1,24 +1,92 @@
 <template>
     <Modal
-        title="Confirm"
-        body="You are about to do something that requires confirmation. Do you wish to continue?"
+        ref="modal"
+        :title="title"
     >
+        <slot>
+            {{ body }}
+        </slot>
+
         <template #footer>
-            <Btn
-                data-bs-dismiss="modal"
-                variant="light"
-            >
-                Cancel
-            </Btn>
-
-            <Btn variant="danger">
-                Confirm
-            </Btn>
+            <slot name="buttonCancel">
+                <Button
+                    class="button-cancel"
+                    :outline="cancelOutline"
+                    :variant="cancelVariant"
+                    @click="cancel"
+                >
+                    {{ cancelText }}
+                </Button>
+            </slot>
+
+            <slot name="buttonOk">
+                <Button
+                    class="button-ok"
+                    :variant="okVariant"
+                    @click="ok"
+                >
+                    {{ okText }}
+                </Button>
+            </slot>
         </template>
     </Modal>
 </template>
 
-<script lang="ts" setup>
-import Btn from '@/components/Button';
+<script lang="ts">
+import {ComponentPublicInstance, ref} from 'vue';
+import Button from '@/components/Button';
 import Modal from '@/components/Modal';
+import {variantProp} from '@/composables/useVariant';
+</script>
+
+<script lang="ts" setup>
+defineProps({
+    body: {
+        type: String,
+        default: 'Are you sure?',
+    },
+    cancelOutline: {
+        type: Boolean,
+        default: false,
+    },
+    cancelText: {
+        type: String,
+        default: 'No',
+    },
+    cancelVariant: variantProp('light'),
+    okText: {
+        type: String,
+        default: 'Yes',
+    },
+    okVariant: variantProp('primary'),
+    title: {
+        type: String,
+        default: 'Confirmation',
+    },
+});
+
+const emit = defineEmits<{
+    (e: 'cancelled'): void,
+    (e: 'confirmed'): void,
+}>();
+
+const cancel = () => {
+    emit('cancelled');
+
+    hide();
+};
+
+const ok = () => {
+    emit('confirmed');
+
+    hide();
+};
+
+const modal = ref<ComponentPublicInstance<typeof Modal>>();
+
+const hide = () => modal.value.bsModal?.hide();
+
+const show = () => modal.value.bsModal?.show();
+
+defineExpose({hide, show});
```

Comment:
Is is also exposing `bsModal` automatically?

Reply:
No, but we exposes `hide, show, toggle` which are all you need in a modal